pkg/signal/testfiles/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"os"
+	"syscall"
+	"time"
+
+	"github.com/docker/docker/pkg/signal"
+)
+
+func main() {
+	sigmap := map[string]os.Signal{
+		"TERM": syscall.SIGTERM,
+		"QUIT": syscall.SIGQUIT,
+		"INT":  os.Interrupt,
+	}
+
+	defer time.Sleep(10 * time.Millisecond)
+
+	signal.Trap(func() {
+		time.Sleep(10 * time.Millisecond)
```

Comment:
I see. I think the `TERM` test is not  testing the correct thing. We actually want the TERM to exit with a specific code, not fallback to `QUIT`, or at least we should test both cases.